graphql-java-servlet/src/main/java/graphql/kickstart/servlet/HttpRequestHandlerImpl.java

```diff
@@ -49,11 +52,31 @@ public void handle(HttpServletRequest request, HttpServletResponse response) thr
   private void execute(GraphQLInvocationInput invocationInput, HttpServletRequest request,
       HttpServletResponse response) {
     try {
+      if (configuration.getResponseCache() != null) {
+        CachedResponse cachedResponse = null;
+        try {
+          cachedResponse = configuration.getResponseCache().get(request, invocationInput);
+        } catch (Throwable t) {
+          log.warn(t.getMessage(), t);
+          log.warn("Ignore read from cache, unexpected error happened");
+        }
+
+        if (cachedResponse != null) {
+          CacheResponseWriter cacheResponseWriter = new CacheResponseWriter();
+          cacheResponseWriter.write(request, response, cachedResponse);
+          return;
+        }
+      }
+
       GraphQLQueryResult queryResult = invoke(invocationInput, request, response);
 
-      QueryResponseWriter queryResponseWriter = QueryResponseWriter.createWriter(queryResult, configuration.getObjectMapper(),
-          configuration.getSubscriptionTimeout());
-      queryResponseWriter.write(request, response);
+      QueryResponseWriter queryResponseWriter = QueryResponseWriter.createWriter(
```

Comment:
I'd add the `ResponseCache` in this factory method instead of adding it to the method signature of the writer.

Reply:
Done.